File: 2006/src/cfortran/Examples/minuit-fcn.f
      Subroutine FCN(Npar,Grad,Fval,Xval,Iflag,Futil)
      Implicit Double Precision (a-h,o-z)

      Dimension Xval(*), Grad(*)

      Data  Xc,Yc/1.11D0,3.14d0/
* 
*  demo FCN 
*  Gunter Folger, December 94
*

       if ( Iflag.eq.1 ) Then
*
*      Initialaise.
*
       Elseif ( Iflag.eq.2) Then
*
*        derivatives...
*
       Endif

       Fval = (Xval(1)-Xc)**2 + (Xval(2)-Yc)**3*Xval(2)

       Return

*      
       End

File: 2006/src/cfortran/Examples/minuit-fcn.c
#include <math.h>
#include <string.h>
#include <cfortran.h>

/* Authors: William Hanlon <whanlon@cosmic.utah.edu>,
 *          Gunter Folger <Gunter.Folger@cern.ch>
 */

/* prototype for benefit of the wrapper */
void fcn(int npar, double grad[3], double * fcnval,
         double xval[2],int iflag, void (*Dummy)());

/* this macro creates a wrapper function called fcn_, which in turn calls the
 * function fcn defined below. The wrapper properly passes ints by value to
 * fcn, while it receives ints by reference from the fortran calling routine.
 */
FCALLSCSUB6(fcn,FCN,fcn,INT,DOUBLEV,PDOUBLE,DOUBLEV,INT,ROUTINE)


void fcn(int npar, double grad[3], double * fcnval,
         double xval[2],int iflag, void (*Dummy)())
{
   double Xc=1.11,Yc=3.14;

   switch(iflag) {
      case 1:
/*
 *      Initialise.
 */
         printf(" fcn_c called to initialise\n");
         break;
      case 2:
/*
 *        derivatives...
 */
      break;

      default:
         *fcnval = pow(xval[0]-Xc,2.) + pow((xval[1]-Yc),3.)*xval[1];
         break;
   }
}
